### PR TITLE
Refactor MediaSession functions to remove session end closure

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Refactor MediaSession, MediaRealTimeSession, and MediaEventProcessor to bring implementation in line with Android implementation.

Remove "sessionEndHandler" closure from Media Session and replace with better logging in MediaRealTimeSession and a bulk remove remove function in MediaEventProcessor to remove sessions which are not active and have no queued events.  Adds `getQueueSize()` function to MediaSession to get the current queue size, which is used by MediaEventProcessor, along with the existing `isMediaSessionActive` to determine if a session can be removed from the processor.

Removes trackerSessionId and dispatchQueue from MediaSession as they are not used. 
Removes trackerSessionId and trackerConfig from MediaEventProcessor.createSession() as they are not used.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Sources/MediaEventProcessor.swift [hunk 1]
```diff
@@ -145,11 +137,17 @@ class MediaEventProcessor: MediaEventProcessing {
             return
         }
 
-        session.abort {
-            self.mediaSessions.removeValue(forKey: sessionId)
-            Log.trace(label: Self.LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - Successfully aborted media session (\(sessionId)).")
-        }
+        session.abort()
+        self.mediaSessions.removeValue(forKey: sessionId)
```
Comment (addb) [inline Sources/MediaEventProcessor.swift:141]: Can we have the logic to remove all the inactive sessions. So we clear out any other session which maybe inactive nut not deleted as we do on the Android side.
`self.mediaSessions.forEach(self.removeInactiveSession)`

**Comment by addb** [comment]
Just one comment. Rest everything looks good!

File: Sources/MediaEventProcessor.swift (hunk 1 above)
Comment (kevinlind) [reply Sources/MediaEventProcessor.swift:141]: It actually is removing all sessions. This private `abort` function is called from `abortAllSessions` which loops through the `mediaSessions` dictionary to abort and remove all sessions.

**Review by addb — APPROVED** [review]
looks good 💯

**Pull request merged**